src/widgets/dialogs/UserInfoPopup.cpp

```diff
@@ -227,6 +228,33 @@ void UserInfoPopup::installEvents()
                                           });
             }
         });
+
+    // ignore highlights
+    QObject::connect(
+        this->ui_.ignoreHighlights, &QCheckBox::clicked, [this](bool gotClicked) mutable {
+            this->ui_.ignoreHighlights->setEnabled(false);
+
+            if (gotClicked) {
+                getApp()->highlights->blacklistedUsers.insertItem(
+                    HighlightBlacklistUser{this->userName_, false});
+                this->ui_.ignoreHighlights->setEnabled(true);
+            } else {
+                const auto &ignoreHighlightsUsersVector =
+                    getApp()->highlights->blacklistedUsers.getVector();
+
+                for (int i = 0; i < ignoreHighlightsUsersVector.size(); i++) {
+                    if (this->userName_ == ignoreHighlightsUsersVector[i].getPattern()) {
+                        getApp()->highlights->blacklistedUsers.removeItem(i);
+                        i--;
+                    }
+                }
+                if (getApp()->highlights->blacklistContains(this->userName_)) {
+                    this->ui_.ignoreHighlights->setToolTip("Name matched by regex");
+                } else {
+                    this->ui_.ignoreHighlights->setEnabled(true);
+                }
+            }
+        });
 }
 
 void UserInfoPopup::setData(const QString &name, const ChannelPtr &channel)

```

Comment:
This function is 77 lines long at the point of this PR. Split this function into multiple reasonably sized sub functions.